divi/src/BlockFactory.cpp

```diff
@@ -65,28 +66,31 @@ bool BlockFactory::AppendProofOfStakeToBlock(
     CBlockTemplate& pBlockTemplate)
 {
     CBlock& block = pBlockTemplate.block;
-    CMutableTransaction txCoinStake;
     SetRequiredWork(pBlockTemplate);
     SetBlockTime(block);
 
     // ppcoin: if coinstake available add coinstake tx
     static int64_t nLastCoinStakeSearchTime = GetAdjustedTime(); // only initialized at startup
 
-    unsigned int nTxNewTime = 0;
-    if(coinstakeCreator_->CreateProofOfStake(
-            block.nBits,
-            block.nTime,
-            nLastCoinStakeSearchTime,
-            txCoinStake,
-            nTxNewTime))
-    {
+    CMutableTransaction txCoinStake;
+    if (customCoinstake_ != nullptr) {
+        txCoinStake = *customCoinstake_;
+    } else {
+        unsigned int nTxNewTime = 0;
+        if(!coinstakeCreator_->CreateProofOfStake(
+                block.nBits,
+                block.nTime,
+                nLastCoinStakeSearchTime,
+                txCoinStake,
+                nTxNewTime))
+            return false;
+
```

Comment:
Were you able to verify that this works? Seems like the endpoint is assuming the provided coinstake has already met the required difficulty target.

Reply:
It certainly works in the way we need it, `vaultfork.py` exercises this.  On regtest if the coin age is large enough, the target to hit will actually be just so large that *every* hash matches.